### ISSUE TITLE
Question about command's auto-complete mechanism

### ISSUE THREAD

**Issue by jslu0418** [purcell/emacs.d#483]
Hi, I am neophyte of Emacs, and I started using this configuration on April 2016. At that time, it's very easy to find command after typing M-X. I thought it's something like auto-full-complete the command at that time. I can just type a few key words of the command and it could show me the thing which exactly I want. But I didn't remember since what time, the command's completion mechanism was changed. It request me to input keywords more strictly. I was confused by it for so long time without asking about it. However, I thing I should know what made this change and how can I modify it to previous. Then maybe I could understand a little more about Emacs than before.

Thank you,

Jiashuai Lu

**Comment by purcell** [comment]
Hi @jslu0418! Yes, I changed from `ido` to `ivy` a while ago, and you can read about `ivy` to get a full understanding. But here's what you need to know: the matching now is essentially a regular expression, and spaces expand to `.*`, ie. "anything". So to run the command `package-list-packages` with `M-x`, previously with `ido` you could have typed `packlist` or `listpack`. Now, with `ivy`, you would need to type `pack list` or ` list pack`. A little retraining is necessary, then, but I've found that quite easy and I recommend you try it too.

(If you really, really can't retrain yourself, you can try calling `sanityinc/enable-ivy-flx-matching`, which should make the matching work more like `ido`, but it might also slow down completion sometimes.)

**Comment by jslu0418** [comment]
Hi Purcell, Thank you for your prompt reply, yeah I tried it just now. It's really easy for me to use too. I am really sorry for bothering you for this small question. I found the new mechanism kind of respect the regular expression, but I find it's quite annoying for typing too much .* and .* . Since you told me the 'space' expand to .* . I feel the convenient world is back, you really made my day. I may still should read about ivy to learn more about it.

Thank you again, you are soooooooooo nice!!!
